src/units/attack_type.cpp

```diff
@@ -97,6 +97,18 @@ std::string attack_type::accuracy_parry_description() const
 	return s.str();
 }
 
+void attack_type::remove_special_by_attribute(const config& filter)
```

Comment:
Shouldn't this be called `remove_special_by_filter`? Since that looks like what it's actually doing.

Reply:
it have remove_ability_by_attribute in unit.cpp, i reused similar name

Reply:
Does `remove_ability_by_attribute` also use a filter, then?

Reply:
Yes, it use"filter_ability" inside remove_ability effect

Reply:
`remove_ability_by_attribute` is a similarly bad name then.

Reply:
> Shouldn't this be called `remove_specials_by_filter`? Since that looks like what it's actually doing, and it potentially removes more than one.

done